### DISCUSSION TITLE
Add CORS Header

### DISCUSSION THREAD

**Discussion by tvogt** [swaggest/rest#81] [Q&A] [answered]
I can't for the life of me figure out where to add a header cleanly to set up CORS. I'm opening this discussion because I think the documentation would benefit from an example for this very common case.

I'm using the related usecase package as well, so the simple outputheaders example here doesn't work for me. All other attempts I've made also didn't work (or overwrite the actual output I want).

**Comment by vearutop** [comment, marked as answer]
Hi, I think the best place to handle CORS would be in a middleware.

Please check [example](https://github.com/swaggest/rest/blob/d3230eb14c8e51d9f331836032dfd73c1c62c4e2/_examples/advanced-generic/router.go#L90-L92).

**Reply by tvogt** [reply]
Agreed. We solved it this way. It's a common case so maybe an example could be in the documentation?
